plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreSchema.java

```diff
@@ -1099,15 +1099,28 @@ public ProceduresCache() {
             super();
         }
 
+        private Boolean xmin_supported;
+
         @NotNull
         @Override
         public JDBCStatement prepareLookupStatement(@NotNull JDBCSession session, @NotNull PostgreSchema owner, @Nullable PostgreProcedure object, @Nullable String objectName) throws SQLException {
             PostgreServerExtension serverType = owner.getDataSource().getServerType();
             String oidColumn = serverType.getProceduresOidColumn(); // Hack for Redshift SP support
+            if (xmin_supported == null) { // Check if xmin supported
+                xmin_supported = false;
+                try (JDBCPreparedStatement dbXminCheck = session.prepareStatement(
+                        "SELECT xmin FROM pg_catalog." + serverType.getProceduresSystemTable() + " limit 1")) {
+                    try (JDBCResultSet dbResult = dbXminCheck.executeQuery()) {
+                        while (dbResult.next()) {
+                            xmin_supported = true;
+                        }
+                    }
+                }
```

Comment:
You're missing a `catch` block here. An exception will be thrown if no such column is present in the result set.

Reply:
You can verify that by changing the `xmin` column to something that is not present in the result set.